lib/ansible/module_utils/network/skydive/skydive.py

```diff
@@ -0,0 +1,171 @@
+        username = provider['username']
+        password = provider['password']
+
+        super(skydive_flow_topology, self).__init__(host, username, password)
+
+    def run(self, ib_spec):
+        state = self.module.params['state']
+        if state not in ('present', 'absent'):
+            self.module.fail_json(msg='state must be one of `present`, `absent`, got `%s`' % state)
+
+        result = {'changed': False}
+        obj_filter = dict([(k, self.module.params[k]) for k, v in iteritems(ib_spec) if v.get('ib_req')])
+
+        proposed_object = {}
+        for key, value in iteritems(ib_spec):
+            if self.module.params[key] is not None:
+                proposed_object[key] = self.module.params[key]
+
+        if state == 'present':
+            capture_str = SKYDIVE_LOOKUP_QUERY + "('Name', '{0}', 'Type', '{1}', 'extra_tcp_metric', '{2}'\
+                                                 ,'ip_defrag', '{3}', 'reassemble_tcp',\
+                                                 '{4}')".format(obj_filter['name'],
+                                                                obj_filter['capture_type'],
+                                                                obj_filter['extra_tcp_metric'],
+                                                                obj_filter['ip_defrag'],
+                                                                obj_filter['reassemble_tcp'])
+            try:
+                self.restclient_object.capture_create(capture_str)
+            except Exception as e:
+                self.module.fail_json(msg=to_text(e))
+            result['changed'] = True
+        if state == 'absent':
+            capture_str = SKYDIVE_LOOKUP_QUERY + "('Name', '{0}', 'Type', '{1}', 'extra_tcp_metric', '{2}'\
+                                                 ,'ip_defrag', '{3}', 'reassemble_tcp',\
+                                                 '{4}')".format(obj_filter['name'],
+                                                                obj_filter['capture_type'],
+                                                                obj_filter['extra_tcp_metric'],
+                                                                obj_filter['ip_defrag'],
+                                                                obj_filter['reassemble_tcp'])
+            # iterating over the captured list and get capture uuid
+            for each_capture in self.restclient_object.capture_list().keys():
+                if capture_str == self.restclient_object.capture_list()[each_capture]['GremlinQuery']:
+                    uuid = each_capture
+                    break
+            try:
+                if uuid:
+                    self.restclient_object.capture_delete(uuid)
+            except Exception:
+                self.module.fail_json(msg="Capture UUID for the input Gremlin query is not found!")
+            result['changed'] = True
+
+        return result
+
+
+class skydive_graph_engine(skydive_base):
```

Comment:
I suspect it is more or less a placeholder. It does not do what it pretends to do. This is where you should use the WSClient

Reply:
I shall create a new PR with WSClient support which will ultimately support node and edge features